ownership-chain/runtime/src/precompiles/mod.rs

```diff
@@ -64,6 +64,8 @@ where
 			a if a == hash(1026) => Some(LivingAssetsPrecompile::execute(handle)),
 			a if a == hash(1027) => Some(LaosEvolution::execute(handle)),
 			a if is_collection_address(a) => Some(Erc721::execute(handle)),
+			a if matches!(address_to_collection_id(a), Ok(_)) =>
```

Comment:
The is_ok() method is specifically designed for this purpose, so it's more clear and concise than using the matches! macro. The matches! macro is more useful when you want to match complex patterns, not just check if a Result is Ok.
```suggestion
			a if address_to_collection_id(a).is_ok() =>
```